generator/generator_tests/node_mixer_test.cpp

```diff
@@ -0,0 +1,57 @@
+#include "testing/testing.hpp"
+
+#include "generator/node_mixer.hpp"
+#include "generator/osm_element.hpp"
+
+#include <sstream>
+
+UNIT_TEST(NodeMixerTests)
+{
+  std::istringstream stream1("");
+  generator::MixFakeNodes(stream1, [](OsmElement * p) {
+    TEST(false, ("Returned an object for an empty input stream."));
+  });
+
+  std::istringstream stream2("shop=gift\nname=Shop\n");
```

Comment:
nit: consider using multi-line string literals

Reply:
Started doing that and understood it hurts readability and introduces indentation issues. So I've decided to leave these strings as is.